prov/verbs/src/fi_verbs.h

```diff
@@ -430,6 +430,7 @@ struct vrb_mem_desc {
 	/* this field is used only by MR cache operations */
 	struct ofi_mr_entry	*entry;
 	enum fi_hmem_iface	iface;
+	uint64_t		device;
```

Comment:
struct ofi_mr_entry is only valid if the MR cache is being used. So we cannot rely on using the information in that structure.

Should this be ofi_mr_hmem_info or ofi_mr_entry? ofi_mr_hmem_info  does haven't enough info, and ofi_mr_entry would have a redundant iov structure since this is stored in ibv_mr.

Reply:
I was suggesting 2 different alternative:

1. convert u64 device to union ofi_mr_hmem_info, which just stores the device in an easier to understand format.

2. Or we always have struct ofi_mr_entry, rather than duplicating the data.  My guess is that once the cache is working properly, it will be enabled more often than not.  And if it is enabled, then we're duplicating the data.

Reply:
As union ofi_mr_hmem_info is defined today, it is missing fields. It could has cuda defined. So it will need to be expanded before it can be used.

We can switch to switch to struct ofi_mr_entry, but that assumes that verbs would only every support a sign IOV which is probably a safe assumption.

Reply:
I'm going to update to use struct ofi_mr_info.